staging/src/k8s.io/apiserver/pkg/storage/feature/feature_support_checker.go

```diff
@@ -82,26 +84,50 @@ func (f *defaultFeatureSupportChecker) Supports(feature storage.Feature) (bool,
 func (f *defaultFeatureSupportChecker) CheckClient(ctx context.Context, c client, feature storage.Feature) error {
 	switch feature {
 	case storage.RequestWatchProgress:
-		return f.clientSupportsRequestWatchProgress(ctx, c)
+		return f.checkClient(ctx, c)
 	default:
 		return fmt.Errorf("feature %q is not implemented in DefaultFeatureSupportChecker", feature)
 
 	}
 }
 
-func (f *defaultFeatureSupportChecker) clientSupportsRequestWatchProgress(ctx context.Context, c client) error {
+func (f *defaultFeatureSupportChecker) checkClient(ctx context.Context, c client) error {
+	go func() {
```

Comment:
With tens of resources, this function will be called many times with the same endpoints. Every time we will spina goroutine. If etcd is unavailable we will have tens of concurrent backoff executions. Instead of hitting etcd 16 times for each endpoint, we will hit it many more times.

The solution would be to spin only one goroutine per endpoint. This can be done by having a additional map to store which endpoints we were already started gorutine for.

Reply:
That is a good idea i'll do that.

Reply:
Sorry one more question. We have caching per endpoint `progresNotifyEndpointCache` which returns fast if the value was cached before, did you consider that too?

Reply:
I thought about this and i think we don't need it, maybe i'm wrong but we have a lock on checking each endpoint and each endpoint will get it's result of feature support, saved in `progresNotifyEndpointCache` and after that the result will return. Adding another map just complicate things up. Please correct me if i'm wrong.

Reply:
We need it because cache `progresNotifyEndpointCache` will not be updated if request fails.